File: Zestawy Zadań/C++/5 - Funkcje, rekurencja/Zestaw 2 zadania/5.17 Dany jest ciąg {W n }.cpp
#include <iostream>

double S3(int n) {
    if (n == 1) {
        return 1;
    } else {
        return 1.0 / (n * (n + 2)) + S3(n - 1);
    }
}

int main() {
    int N;
    std::cout << "Podaj liczbê ca³kowit¹: ";
    std::cin >> N;
    std::cout << "Wynik: " << S3(N) << std::endl;
    return 0;
}

File: Zestawy Zadań/Python/5 - Funkcje, rekurencja/Zestaw 1 zadania/5.17 Dany jest ciąg {W n }.py
def S3(n):
    if n == 1:
        return 1
    else:
        return 1 / (n * (n + 2)) + S3(n - 1)

if __name__ == "__main__":
    N = int(input("Podaj liczbę całkowitą: "))
    print("Wynik:", S3(N))
